cmd/soroban-cli/src/rpc/transaction.rs

```diff
@@ -84,6 +92,189 @@ pub fn assemble(
+    // TODO: Deal with potentual conversion errors here
+    let tx_resources = TransactionResources {
+        instructions: resources.instructions,
+        read_entries: resources.footprint.read_only.len().try_into().unwrap(),
+        write_entries: resources.footprint.read_write.len().try_into().unwrap(),
+        read_bytes: resources.read_bytes,
+        write_bytes: resources.write_bytes,
+        metadata_size_bytes: resources.extended_meta_data_size_bytes,
+        transaction_size_bytes: envelope.to_xdr()?.len().try_into().unwrap(),
+    };
+
+    let (fee, new_refundable_fee) =
+        compute_transaction_resource_fee(&tx_resources, fee_configuration);
+    let mut tx = raw.clone();
+    // TODO: Deal with this error
+    tx.fee = tx.fee.max((fee * 115 / 100).try_into().unwrap());
+    tx.ext = TransactionExt::V1(SorobanTransactionData {
+        ext: ext.clone(),
+        resources: resources.clone(),
+        refundable_fee: new_refundable_fee,
+    });
+    Ok(tx)
+}
+
+// Use the given source_key and signers, to sign all SorobanAuthorizationEntry's in the given
+// transaction. If unable to sign, return an error.
+pub fn sign_soroban_authorizations(
+    raw: &Transaction,
+    source_key: &ed25519_dalek::Keypair,
+    signers: &[ed25519_dalek::Keypair],
+    signature_expiration_ledger: u32,
+    network_passphrase: &str,
+) -> Result<(Transaction, Vec<SorobanAuthorizationEntry>), Error> {
+    let mut tx = raw.clone();
+
+    if tx.operations.len() != 1 {
+        // This must not be an invokeHostFunction operation, so nothing to do
+        return Ok((tx, Vec::new()));
+    }
+
+    let mut op = tx.operations[0].clone();
+    let OperationBody::InvokeHostFunction(ref mut body) = &mut op.body else {
+        return Ok((tx, Vec::new()));
+    };
+
+    let network_id = Hash(Sha256::digest(network_passphrase.as_bytes()).into());
+    // TODO: Pass this in from the caller? Or where do we get this from? Can we just use the one
+    // already in the auth?
+    // let ledger_validity_count = 5;
+
+    let source_address = source_key.public.as_bytes();
+
+    let signed_auths = body
+        .auth
+        .iter()
+        .map(|raw_auth| {
+            let mut auth = raw_auth.clone();
+            let SorobanAuthorizationEntry {
+                credentials: SorobanCredentials::Address(ref mut credentials),
+                ..
+            } = auth else {
+                // Doesn't need special signing
+                return Ok(auth);
+            };
+            let SorobanAddressCredentials { ref address, .. } = credentials;
+
+            // See if we have a signer for this authorizationEntry
+            // If not, then we Error
+            let needle = match address {
+                ScAddress::Account(AccountId(PublicKey::PublicKeyTypeEd25519(Uint256(ref a)))) => a,
+                ScAddress::Contract(Hash(c)) => {
+                    // This address is for a contract. There's no way to sign it because it is malformed.
```

Comment:
FWIW contract address is not 'malformed', it just means that a custom account is supposed to be used. It's fine that we don't support this initially, but we probably should say that this is not supported yet, not that something is wrong